arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentCollectionViewController.swift

```diff
@@ -89,12 +89,20 @@ class ContentCollectionViewController: UICollectionViewController, UICollectionV
     // MARK: - UICollectionViewDelegate
     
     override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        // hide keyboard if visible
+        // Hide keyboard if visible.
         view.endEditing(true)
-        
         let category = categories[indexPath.item]
         let controller = storyboard!.instantiateViewController(withIdentifier: "ContentTableViewController") as! ContentTableViewController
-        controller.allSamples = category.samples
+        // Filter and display the favorited samples.
+        if category.name == "Favorites" {
+            let allSamples = categories.flatMap { $0.samples }
+            let allFavorites = allSamples.filter { $0.isFavorite }
```

Comment:
`ContentTableViewController.swift` is also performing a `.filter { $0.isFavorite }`.  Can one of them be removed?  If you do the filter here, is `ContentTableViewController.isFavoritesCategory` really necessary?

Reply:
Yes, both are necessary. Unfortunately, there's no simple way to generalize both cases. Both portions load the favorited samples but each are part of an existing process to handle all the categories.